packages/domains-table/src/domains-table/domains-table-site-cell.tsx

```diff
@@ -0,0 +1,57 @@
+import { DomainData, SiteDetails } from '@automattic/data-stores';
+import { createInterpolateElement } from '@wordpress/element';
+import { useI18n } from '@wordpress/react-i18n';
+
+interface DomainsTableSiteCellProps {
+	site?: Pick< SiteDetails, 'ID' | 'name' >;
+	currentDomainData?: DomainData;
+	siteSlug: string;
+}
+
+export const DomainsTableSiteCell = ( {
+	site,
+	currentDomainData,
+	siteSlug,
+}: DomainsTableSiteCellProps ) => {
+	const { __ } = useI18n();
+
+	if ( ! site || ! currentDomainData ) {
+		return null;
+	}
+
+	if ( currentDomainData.current_user_can_create_site_from_domain_only ) {
+		return createInterpolateElement(
+			/* translators: ariaHidden means that the component will be skipped by screen readers. */
+			__(
+				'<create>Create</create> <ariaHidden>or</ariaHidden> <connect>connect</connect> <ariaHidden>a site</ariaHidden>'
```

Comment:
Why would you want to hide the sentences from users with screen readers? I'm sure they're used to the idea of having having links embedded in sentences.

Reply:
They do, but this experience is better. When I switch focus, on VoiceOver at least, I don't get whole sentence read out loud. I get the link focused and it only says "Create" and the next tab, VoiceOver reads as "Connect".

Hiding sentences and attributing `aria-label`s allow us to come with a better screen reading experience.

Reply:
Screen readers aren't necessarily just navigating by jumping focus around. There's keyboard controls for navigating paragraph by paragraph, or horizontally/vertically through table cells. I'm thinking of that later case. When a sighted person reads the cell text it will say "Create or connect a site". When a screenreader user reads the cell it will say "Create connect". They'll also know it's two links so they won't be completely lost. It's just that if we think the sentence is useful for sighted people, why wouldn't it be just as useful for screenreader users?

Maybe we don't need the sentence at all? I notice the link in the mockup is just "Create site", maybe we should go with that?

Reply:
>Maybe we don't need the sentence at all? I notice the link in the mockup is just "Create site", maybe we should go with that?

That's something I'd like @JanaMW27's input. I don't think we should drop the "Connect a site" capability, so that's why I didn't follow the "Create a site" mockup entirely.

Reply:
Just to be clear this is the "connect a site" we're talking about

![CleanShot 2023-08-30 at 10 16 50@2x](https://github.com/Automattic/wp-calypso/assets/1500769/edde138e-85d0-4833-84c2-59f52a4775f9)